### PR TITLE
field-array comparisons in associations should be rewritten as f op v…

### PR DESCRIPTION
**Pull request by bserdar**
…1 or f op v2...

### CODE REVIEW

File: crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java [hunk 1]
```diff
@@ -1076,4 +1076,14 @@ public void test_self_ref_array_not_contains() throws Exception {
...
     }
+
+    @Test
+    public void test_self_ref_with_parents() throws Exception {
+        FindRequest fr=new FindRequest();
+        fr.setQuery(query("{'field':'_id','op':'=','rvalue':'img1'}"));
+        fr.setProjection(projection("[{'field':'*','recursive':true},{'field':'vulnerabilities.*.packages.*.fixed_by_images.*'}]"));
+        fr.setEntityVersion(new EntityVersion("containerImage-self","0.0.1"));
+        Response response=mediator.find(fr);
+        System.out.println(response.getEntityData());
```
Comment (BVulaj) [inline crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java]: Can we assert anything?

Comment (bserdar) [inline crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java]: oh, I forgot...

**Comment by bserdar** [comment]
done.

**Comment by bserdar** [comment]
Done.

**Pull request merged**